cdap-ui/cypress/integration/securekeymanager.spec.ts

```diff
@@ -0,0 +1,155 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as Helpers from '../helpers';
+let headers = {};
+
+const { dataCy } = Helpers;
+
+const MOCK_SECURE_KEYS: any[] = [
+  {
+    name: 'secure-key-1-id',
+    description: 'Example Secure Key 1',
+    data: 'WARNING: this is secure data',
+    createdEpochMs: 1471718010326,
+  },
+  {
+    name: 'secure-key-2-id',
+    description: 'Example Secure Key 2',
+    data: 'WARNING: this is secure data',
+    createdEpochMs: 1471718010327,
+  },
+  {
+    name: 'secure-key-3-id',
+    description: 'Example Secure Key 3',
+    data: 'WARNING: this is secure data',
+    createdEpochMs: 1471718010328,
+  },
+];
+
+describe('Secure Key Manager Page', () => {
+  // Uses API call to login instead of logging in manually through UI
+  before(() => {
+    Helpers.loginIfRequired().then(() => {
+      cy.getCookie('CDAP_Auth_Token').then((cookie) => {
+        if (!cookie) {
+          return;
+        }
+        headers = {
+          Authorization: 'Bearer ' + cookie.value,
+        };
+      });
+    });
+    const stub = cy.stub();
+    cy.window().then((win) => {
+      win.onbeforeunload = null;
+    });
+    cy.on('window:confirm', stub);
+
+    Helpers.getArtifactsPoll(headers);
+  });
+
+  describe('Accessing and managing secure keys', () => {
+    before(() => {
+      cy.visit('/cdap/ns/default/securekeys');
+      cy.wait(6000); // wait for secure keys to be loaded
+    });
+
+    it('should delete MOCK_SECURE_KEYS before testing if they already exist in the secure storage', () => {
+      cy.get('body').then((body) => {
+        MOCK_SECURE_KEYS.forEach((key) => {
+          const secureKeyMenu = `${dataCy(`secure-key-row-${key.name}`)} ${dataCy(`menu-icon`)}`;
+          if (body.find(secureKeyMenu).length > 0) {
+            cy.get(secureKeyMenu).click();
+
+            // open a delete dialog
+            cy.get(dataCy('delete-secure-key')).click();
+
+            // confirm delete
+            cy.get(dataCy('Delete')).click();
+            cy.wait(6000); // wait for success alert component to disappear
+          }
+        });
+      });
+    });
+
+    it('should add secure keys', () => {
+      MOCK_SECURE_KEYS.forEach((key) => {
+        cy.get(dataCy('create-secure-key')).click({ force: true });
+
+        cy.get(dataCy('secure-key-name'))
+          .click()
+          .type(key.name);
+        cy.get(dataCy('secure-key-description'))
+          .click()
+          .type(key.description);
+        cy.get(dataCy('secure-key-data'))
+          .click()
+          .type(key.data);
+
+        cy.get(dataCy('save-secure-key')).click();
+        cy.wait(6000); // wait for success alert component to disappear
```

Comment:
same as above

Reply:
Put comment above regarding alternative for cy.wait(6000);